x-pack/plugins/ml/public/application/util/custom_url_utils.ts

```diff
@@ -184,11 +184,13 @@ function buildKibanaUrl(urlConfig: UrlConfig, record: CustomUrlAnomalyRecordDoc)
 
           let availableCharactersLeft =
             URL_LENGTH_LIMIT - resultPrefix.length - resultPostfix.length;
-          const queryFields = queryString
+
+          // URL template might contain encoded characters
+          const queryFields = decodeURIComponent(queryString)
             // Split query string by AND operator.
             .split(/\sand\s/i)
             // Get property name from `influencerField:$influencerField$` string.
-            .map((v) => v.split(':')[0]);
+            .map((v) => String(v.split(/:(.+)?\$/)[0]).trim());
```

Comment:
Tested a custom URL on the special characters data set and am seeing this error when opening a link to Dashboard:

![image](https://user-images.githubusercontent.com/7405507/92628113-39798b00-f2c4-11ea-8a71-2a9274708818.png)

Job config has:
```
  "analysis_config": {
    "bucket_span": "15m",
    "detectors": [
      {
        "detector_description": "mean(\"metric%$£&!{(]field\") partitionfield=\"at@name\"",
        "function": "mean",
        "field_name": "metric%$£&!{(]field",
        "partition_field_name": "at@name",
        "detector_index": 0
      }
    ],
    "influencers": [
      "at@name",
      "singlequote'name"
    ]
  },
```

And opening a URL which uses the influencers from this anomaly:
![image](https://user-images.githubusercontent.com/7405507/92628344-92e1ba00-f2c4-11ea-83a3-da52065602c5.png)

I think it is the `singlequote'name` field name which is causing the problem here.

Reply:
Fixed in https://github.com/elastic/kibana/pull/76957/commits/57fa5a00eb286758dff6bcf7b592f10dd5399905